### PR TITLE
hyperscan: better error message if not compiled

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

File: src/detect-engine-mpm.c [hunk 1]
```diff
@@ -643,6 +643,13 @@ uint16_t PatternMatchDefaultMatcher(void)
             }
         }
 
+#ifndef BUILD_HYPERSCAN
+    if (strcmp(mpm_algo, "hs") == 0) {
+        FatalError(SC_ERR_INVALID_VALUE, "Hyperscan (hs) support for mpm-algo is "
+                   "not compiled into Suricata. Add --with-libhs-includes and "
+                   "--with-libhs-libraries to configure.");
```
Comment (jasonish) [inline src/detect-engine-mpm.c:650]: I can live with this error. But I find in most cases if Hyperscan is simply installed, Suricata will pick it up without these options.

File: src/detect-engine-mpm.c [hunk 2]
```diff
@@ -643,6 +643,13 @@ uint16_t PatternMatchDefaultMatcher(void)
             }
         }
 
+#ifndef BUILD_HYPERSCAN
+    if (strcmp(mpm_algo, "hs") == 0) {
```
Comment (victorjulien) [inline src/detect-engine-mpm.c:647]: scan-build is unhappy here
```
   CC       detect-engine-mpm.o
detect-engine-mpm.c:647:9: warning: Null pointer passed to 1st parameter expecting 'nonnull'
    if (strcmp(mpm_algo, "hs") == 0) {
        ^~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

File: src/util-spm.c [hunk 3]
```diff
@@ -83,6 +83,13 @@ uint16_t SinglePatternMatchDefaultMatcher(void)
             }
         }
 
+#ifndef BUILD_HYPERSCAN
+        if (strcmp(spm_algo, "hs") == 0) {
```
Comment (victorjulien) [inline src/util-spm.c:87]:
```
   CC       util-spm.o
util-spm.c:87:13: warning: Null pointer passed to 1st parameter expecting 'nonnull'
        if (strcmp(spm_algo, "hs") == 0) {
            ^~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

**Comment by inashivb** [comment]
Replaced with #5151

**Pull request closed without merging**